### PR TITLE
Ascendex fetchClosedOrders() Swap Functionality

### PR DESCRIPTION
**Pull request by Dan-krm**
Added swap functionality for fetchClosedOrders() Private Method:
```
ascendex.fetchClosedOrders (BTC/USDT:USDT)
1389 ms
                              id | clientOrderId | timestamp | datetime | lastTradeTimestamp |        symbol |  type | timeInForce | postOnly | side | price | stopPrice | amount |    cost | average | filled | remaining |   status |                                   fee | trades |                                    fees
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
r17de5dca15fU0711043490bbtcp0nQX |               |           |          |      1640239341427 | BTC/USDT:USDT | limit |             |          |  buy | 30000 |         0 |  0.001 |       0 |         |      0 |     0.001 | canceled |          {"cost":0,"currency":"USDT"} |     [] |          [{"cost":0,"currency":"USDT"}]
r17de4e52676U0711043490bbtcp6n3l |               |           |          |      1640222945863 | BTC/USDT:USDT | limit |             |          |  buy | 42000 |         0 | 0.0007 |       0 |         |      0 |    0.0007 | canceled |          {"cost":0,"currency":"USDT"} |     [] |          [{"cost":0,"currency":"USDT"}]
r17de4946cb0U0711043490bbtcpRYea |               |           |          |      1640217472717 | BTC/USDT:USDT | limit |             |          |  buy | 48591 |         0 | 0.0016 | 77.7456 |         | 0.0016 |         0 |   closed | {"cost":0.04664736,"currency":"USDT"} |     [] | [{"cost":0.04664736,"currency":"USDT"}]
r17de491d7f1U0711043490bbtcpIU5L |               |           |          |      1640217426062 | BTC/USDT:USDT | limit |             |          |  buy | 42588 |         0 | 0.0018 |       0 |         |      0 |    0.0018 | canceled |          {"cost":0,"currency":"USDT"} |     [] |          [{"cost":0,"currency":"USDT"}]
4 objects
```

### CODE REVIEW

File: js/ascendex.js [hunk 1]
```diff
@@ -1289,7 +1289,10 @@ module.exports = class ascendex extends Exchange {
             market = this.market (symbol);
             request['symbol'] = market['id'];
         }
-        const method = this.safeValue (options, 'method', 'v1PrivateAccountGroupGetOrderHist');
+        let method = this.safeValue (options, 'method', 'v1PrivateAccountGroupGetOrderHist');
+        if (market['swap']) {
```
Comment (kroitor) [inline js/ascendex.js:1293]: The `market` is not always defined here.

Comment (kroitor) [reply js/ascendex.js:1293]: See how it's done here: https://github.com/ccxt/ccxt/blob/master/js/okex.js#L2062

File: js/ascendex.js [hunk 2]
```diff
@@ -1266,7 +1267,7 @@ module.exports = class ascendex extends Exchange {
         if (symbol !== undefined) {
             market = this.market (symbol);
         }
-        const defaultAccountCategory = this.safeString (this.options, 'account-category');
+        const defaultAccountCategory = this.safeString (this.options, 'account-category', this.defaultType);
```
Comment (kroitor) [inline js/ascendex.js:1270]: There's no such thing as `this.defaultType`, but there is `this.options['defaultType']` and it's already handled in handleMarketTypeAndParams: 

- https://github.com/ccxt/ccxt/blob/master/js/base/Exchange.js#L2131
- https://github.com/ccxt/ccxt/blob/master/python/ccxt/base/exchange.py#L2720
- https://github.com/ccxt/ccxt/blob/master/php/Exchange.php#L3599

**Comment by Dan-krm** [comment]
On line 1290 I'm not sure how to fix this error that happens if I switch 'fetchOpenOrders' to 'fetchClosedOrders':
[NotSupported] ascendex [object Object] does not have a value in mapping

File: js/ascendex.js [hunk 3]
```diff
@@ -1263,7 +1263,7 @@ module.exports = class ascendex extends Exchange {
     async fetchClosedOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
         await this.loadMarkets ();
         await this.loadAccounts ();
-        const defaultAccountCategory = this.safeString (this.options, 'account-category', this.defaultType);
+        const defaultAccountCategory = this.safeString (this.options, 'account-category');
```
Comment (kroitor) [inline js/ascendex.js:1266]: Technically, the defaultAccountCategory must be reworked to defaultType, we should only use handleMarketTypeAndParams here, replacing the old account-category-code with it.

File: js/ascendex.js [hunk 4]
```diff
@@ -1285,11 +1282,16 @@ module.exports = class ascendex extends Exchange {
             // 'page': 1,
             // 'pageSize': 100,
         };
+        let market = undefined;
         if (symbol !== undefined) {
             market = this.market (symbol);
             request['symbol'] = market['id'];
         }
-        const method = this.safeValue (options, 'method', 'v1PrivateAccountGroupGetOrderHist');
+        const [ methodType, query ] = this.handleMarketTypeAndParams ('fetchOpenOrders', market, params);
```
Comment (kroitor) [inline js/ascendex.js:1290]: Should be `fetchClosedOrders` here.

**Comment by kroitor** [comment]
> On line 1290 I'm not sure how to fix this error that happens if I switch 'fetchOpenOrders' to 'fetchClosedOrders': [NotSupported] ascendex [object Object] does not have a value in mapping

How do you reproduce it?

File: js/ascendex.js (hunk 4 above)
Comment (Dan-krm) [reply js/ascendex.js:1290]: On line 1290 I'm not sure how to fix this error that happens if I switch 'fetchOpenOrders' to 'fetchClosedOrders':
[NotSupported] ascendex [object Object] does not have a value in mapping

**Comment by Dan-krm** [comment]
> 

If I change it to 'fetchClosedOrders' then:
node examples/js/cli ascendex fetchClosedOrders

**Comment by kroitor** [comment]
@Dan-krm add `console.log (methodType)` after line 1290. It will print the selected type. That exception is thrown by `getSupportedMapping`.

**Comment by Dan-krm** [comment]
@kroitor 
It says [object Object]

**Comment by kroitor** [comment]
@Dan-krm it must be a string instead – try figuring it out how it could possibly become an object.

**Pull request merged**